src/app-layer-ssl.c

```diff
@@ -541,6 +541,17 @@ static int TlsDecodeHSCertificate(SSLState *ssl_state, SSLStateConnp *connp,
         }
         connp->cert0_issuerdn = str;
 
+        connp->cert0_sans_len = rs_x509_get_subjectaltname_len(x509);
+        char **sans = SCCalloc(connp->cert0_sans_len, sizeof(char *));
+        for (uint8_t i = 0; i < connp->cert0_sans_len; i++) {
+            sans[i] = rs_x509_get_subjectaltname_at(x509, i);
+        }
+        connp->cert0_sans = sans;
```

Comment:
Could this be done in some other way?
The args are available and parsed in Rust but need to make them available in C to use later on to match on sticky buffer.
Thought of having a type in Rust but not sure what that would look like.

Reply:
This is probably an OK option.  The pointers return are stable until free'd, which is in the callers control here.

I guess it also depends on where these are going to be used, for example if you look at `detect-dns-answer-name.c` we request the values from Rust as needed, rather than building a collection first.